tools/pdf2txt.py

```diff
@@ -14,7 +15,190 @@
+        encoding: Output encoding. (default: utf-8)
+        input_file: Pdf file to parse through
+        outtype: A string defines witch type of the file to create.
+    """
+    PDFDocument.debug = debug
+    PDFParser.debug = debug
+    CMapDB.debug = debug
+    PDFPageInterpreter.debug = debug
+    #
+    rsrcmgr = PDFResourceManager(caching=caching)
+    if not outtype:
+        outtype = 'text'
+        if outfile:
+            if outfile.endswith('.htm') or outfile.endswith('.html'):
+                outtype = 'html'
+            elif outfile.endswith('.xml'):
+                outtype = 'xml'
+            elif outfile.endswith('.tag'):
+                outtype = 'tag'
+    if outtype:
+        if outfile:
+            outfp = open(outfile, 'w', encoding=encoding)
+        else:
+            outfp = sys.stdout
+
+        if outtype == 'text':
+            device = TextConverter(rsrcmgr, outfp, laparams=laparams,
+                                   imagewriter=imagewriter)
+        elif outtype == 'xml':
+            device = XMLConverter(rsrcmgr, outfp, laparams=laparams,
+                                  imagewriter=imagewriter,
+                                  stripcontrol=stripcontrol)
+        elif outtype == 'html':
+            device = HTMLConverter(rsrcmgr, outfp, scale=scale,
+                                   layoutmode=layoutmode, laparams=laparams,
+                                   imagewriter=imagewriter, debug=debug)
+        elif outtype == 'tag':
+            device = TagExtractor(rsrcmgr, outfp)
+
+        for fname in input_file:
+            # input_file_name = fname.split('/')[-1]
+            # Get the input file name to create a directory
+            # with all the chapters
+            # input_file_name = input_file_name.replace('.pdf', '')
+
+            with open(fname, 'rb') as fp:
+                interpreter = PDFPageInterpreter(rsrcmgr, device)
+                for page in PDFPage.get_pages(
+                        fp, pagenos, maxpages=maxpages, password=password,
+                        caching=caching, check_extractable=True
+                ):
+                    page.rotate = (page.rotate + rotation) % 360
+                    interpreter.process_page(page)
+        device.close()
+
+        if outfile:
+            outfp.close()
+
+        return True
+
+
+def create_chapters(debug: int,
+                    caching: bool,
+                    outfile: str,
+                    laparams: LAParams,
+                    imagewriter: ImageWriter,
+                    stripcontrol: bool,
+                    scale: int,
+                    layoutmode: str,
+                    pagenos: Set[int],
+                    maxpages: int,
+                    password: str,
+                    rotation: int,
+                    encoding: str,
+                    input_file: List[str],
+                    outtype: str,
+                    chapter_definition: str) -> bool:
```

Comment:
In pdf2txt.py, on line 234 there are default values for most of these parameters. https://github.com/kakann/pdfminer/blob/c53cfd375225efba4e0ed35ec608fe48f1be6c96/tools/pdf2txt.py#L234-L250

Move these default values to this function instead e.g.
```
debug: int = 0,
caching: bool = True
...
```
This way in the tests you don't have to specify every parameter, but instead only specify those that are needed

Reply:
Except for laparams the other values are parsed through with argparser depending on which flags are used, so the variables are going to be needed. We thought about moving laparams into the function but since all the other values are created there so we thought it is easier to find it there.

Reply:
You do actually have a point there. The variables would have to be there either way. Alright they can stay